python/ray/serve/utils.py

```diff
@@ -84,21 +84,24 @@ async def body(self):
 
 
 def parse_request_item(request_item):
-    arg = request_item.args[0] if len(request_item.args) == 1 else None
-
-    # If the input data from handle is web request, we don't need to wrap
-    # it in ServeRequest.
-    if isinstance(arg, starlette.requests.Request):
-        return arg
-    elif isinstance(arg, HTTPRequestWrapper):
-        return build_starlette_request(arg.scope, arg.body)
-    else:
-        return ServeRequest(
-            arg,
-            request_item.kwargs,
-            headers=request_item.metadata.http_headers,
-            method=request_item.metadata.http_method,
-        )
+    if len(request_item.args) <= 1:
+        arg = request_item.args[0] if len(request_item.args) == 1 else None
```

Comment:
```suggestion
    if len(request_item.args) == 1:
        arg = request_item.args[0]
```
because the None case just go back to the last return anyway

Reply:
Ah we need this case for the legacy `ServeRequest` case when the user passes no args to the serve handle